### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by derNarr**
adds section about how to keep fork in sync with upstream on github in misc chapter

This was asked by a collaborator and I think it would not hurt, if we include this. If you think it is unnecessary clutter, let me know.

### CODE REVIEW

**Comment by Trybnetic** [comment]
I think [GitHub's Documentation on forks](https://help.github.com/articles/working-with-forks/) already contains all necessary information.

It might be a good idea to merge the short version into the development section instead of misc.

File: doc/source/misc.rst [hunk 1]
```diff
@@ -72,6 +73,55 @@ Further reading:
 - https://stackoverflow.com/questions/44379525/r-reticulate-how-do-i-clear-a-python-object-from-memory
 
 
+Keeping a fork in sync with master
```
Comment (dekuenstle) [inline doc/source/misc.rst:76]: I agree with @Trybnetic 's comment about this: This is a duplicate of github's [docs](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/syncing-a-fork). 
We simply could link to this in the developer.rst

File: doc/source/misc.rst [hunk 2]
```diff
@@ -99,3 +149,11 @@ Sometimes it is useful to monitory the memory footprint of the python process.
 This can be achieved by using ``memory_profiler``
 (https://pypi.python.org/pypi/memory_profiler).
 
+
+CPU profiling of C extensions
```
Comment (dekuenstle) [inline doc/source/misc.rst:153]: Testing related stuff fits better in the developer.rst in my oppinion.

**Review by dekuenstle — CHANGES_REQUESTED** [review]
Developer related docs should be in the developer section.

File: doc/source/misc.rst (hunk 2 above)
Comment (derNarr) [reply doc/source/misc.rst]: I moved it to developer.rst

File: doc/source/misc.rst (hunk 1 above)
Comment (derNarr) [reply doc/source/misc.rst]: I still feel that I want to have it in for now. It still gives you a good indication what to do without clicking on a link and documents how (I / we) do it.

**Pull request merged**